### PR TITLE
RN navigation

### PR DESCRIPTION
**Pull request by veksi**
Since this chapter contains some code related to navigation structure, I think that @andreicek and @doroz0 should check it out, and make sure this is not just my personal preference.
This is something we should generaly all define and agree to be as an general approach on company level.

### CODE REVIEW

File: React Native/Navigation.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+Routing in mobile applications presents more then mere navigation between routes (screens). It consists of functionalities and styles which are divided across platforms. Generally, routing systems inside native applications has its own look and feel aswell as well defined UX which differs from android and iOS platforms.
```
Comment (nives113) [inline React Native/Navigation.md:1]: `has its own look and feel aswell as well defined UX which differs` should probably be
`has its own look and feel as well defined UX which differs`

File: React Native/Navigation.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+```
+2.
+ ```javascript
+export interface IHomeRouteProps {
+	propName: propValue;
+	...
+};
+```
+3.
+ ```javascript
```
[the remark this answers is not in the document]
Comment (veksi) [reply React Native/Navigation.md:144]: I agree.

File: React Native/Navigation.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
+Routing in mobile applications presents more then mere navigation between routes (screens). It consists of functionalities and styles which are divided across platforms. Generally, routing systems inside native applications has its own look and feel as well as well defined UX which differs from android and iOS platforms.
+
+In order to achieve this “native feel”, RN community implemented a [React Navigation](https://reactnavigation.org/) library. It’s main focus presents “native-look-and-feel” while keeping the performance and extensibility.
+
+### Official documentation
+[Official documentation](https://reactnavigation.org/docs/getting-started) has a lot of informations and contains more then few code examples of all important parts.
+Make sure to get faimiliar with it before continue to read further.
```
Comment (doroz0) [inline React Native/Navigation.md:7]:
```suggestion
Make sure to get familiar with it before continuing to read further.
```

File: React Native/Navigation.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
+Routing in mobile applications presents more then mere navigation between routes (screens). It consists of functionalities and styles which are divided across platforms. Generally, routing systems inside native applications has its own look and feel as well as well defined UX which differs from android and iOS platforms.
+
+In order to achieve this “native feel”, RN community implemented a [React Navigation](https://reactnavigation.org/) library. It’s main focus presents “native-look-and-feel” while keeping the performance and extensibility.
+
+### Official documentation
+[Official documentation](https://reactnavigation.org/docs/getting-started) has a lot of informations and contains more then few code examples of all important parts.
```
Comment (doroz0) [inline React Native/Navigation.md:6]:
```suggestion
[Official documentation](https://reactnavigation.org/docs/getting-started) has a lot of information and contains more than a few code examples of all important parts.
```

File: React Native/Navigation.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+Make sure to get faimiliar with it before continue to read further.
+
+### 1. Navigation file structure
+In order to keep your navigation maintanable and easy to upgrade  here are some guidelines which rely on library main concepts: ***Stack***, ***Tab***, ***Drawer*** and ***Parameters***.
```
Comment (doroz0) [inline React Native/Navigation.md:10]:
```suggestion
In order to keep your navigation maintainable and easy to upgrade, here are some guidelines which rely on library main concepts: ***Stack***, ***Tab***, ***Drawer*** and ***Parameters***.
```

File: React Native/Navigation.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+ - navigation/
+	  -- stacks/
```
Comment (doroz0) [inline React Native/Navigation.md:13]:
```suggestion
	 - stacks/
```

File: React Native/Navigation.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+	  -- tabs/
```
Comment (doroz0) [inline React Native/Navigation.md:14]:
```suggestion
	 - tabs/
```

File: React Native/Navigation.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+	  -- drawer/
```
Comment (doroz0) [inline React Native/Navigation.md:15]:
```suggestion
	 - drawer/
```

File: React Native/Navigation.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+	  -- params/
```
Comment (doroz0) [inline React Native/Navigation.md:16]:
```suggestion
	 - params/
```

File: React Native/Navigation.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+	  -- Navigator.tsx
```
Comment (doroz0) [inline React Native/Navigation.md:17]:
```suggestion
	 - Navigator.tsx
```

File: React Native/Navigation.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+			// NAVIGATION STRUCTURE
+		</NavigationContainer>
+	);
+};
+```
+
+#### 1.2 STACKS
+All navigation stacks should be wrapped as functional components inside `navigation/stacks`.
+For example, most common application structure consits of authentication (*login*, *register*) and main (*home*, etc..) workflow.
+In React Navigation terms this means that you should split your navigation stacks between those 2 flows: `AuthenticationStack` and `HomeStack`. Even though, we could use only one stack with all screens, this is a bad practise in mobile development and makes the app harder to maintain as your application grows.
```
Comment (doroz0) [inline React Native/Navigation.md:37]:
```suggestion
In React Navigation terms this means that you should split your navigation stacks between those 2 flows: `AuthenticationStack` and `HomeStack`. Even though we could use only one stack with all screens, this is a bad practice in mobile development and makes the app harder to maintain as your application grows.
```

File: React Native/Navigation.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+2.
+ ```typescript
+export interface IHomeRouteProps {
+	propName: propValue;
+	...
+};
+```
+3.
+ ```typescript
+export type HomeStackScreenProps<T extends keyof HomeStackParamList = {
```
Comment (RaulSusanjSamolov) [inline React Native/Navigation.md:145]: Needs to be with closing tag:

HomeStackScreenProps<T extends keyof HomeStackParamList **>**

File: React Native/Navigation.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+		<Tab.Navigator>
+			<Tab.Screen component={AuthenticationStack} name="AuthenticationStack" />
+			<Tab.Screen component={HomeStack} name="HomeStack" />
+		</Tab.Navigator>
+	);
+};
+```
+
+#### 1.4 Drawer
+Usually, applications have one navigation tab which should be in `navigation/drawer`. When using tab navigation, each `Tab.Screen` component should target a stack with related screens.
```
Comment (RaulSusanjSamolov) [inline React Native/Navigation.md:104]: Should be Drawer.Screen ?

Comment (veksi) [reply React Native/Navigation.md]: Good catch!

**Review by DarkoKukovec — APPROVED** [review]

**Pull request merged**